### PR TITLE
feat: #1118 NoAuth_AllWays_SelectTableView test

### PR DESCRIPTION
**Pull request by EkaterinaSokolova-V**
Added:
NoAuth_AllWays_SelectTableView test

### CODE REVIEW

**Review by Ekaterina1994 — APPROVED** [review]

File: mw-webapp/cypress/e2e/allWays.cy.ts [hunk 1]
```diff
@@ -8,6 +8,24 @@ describe('All Ways scope tests', () => {
...
+    cy.clearAllStorage();
+  });
+
+  it('NoAuth_AllWays_SelectTableView', () => {
+    const tableHeadersEn = Object.values(allWaysPageContent.waysTable.columns).map(column => column.en);
+    tableHeadersEn.push(allWaysPageData.starSimbol);
+
+    allWaysSelectors.filterViewBlock.getTableViewButton().click();
+
+    allWaysSelectors.allWaysTable.getTable().find('thead tr th div div:first-child').each(($el, index) => {
```
Comment (tritonJS826) [inline mw-webapp/cypress/e2e/allWays.cy.ts:25]: you don't need to use "$" sign. $ related to legacy practice (jquery). Just "el" is better by modern codestyle than "$el"

Comment (tritonJS826) [inline mw-webapp/cypress/e2e/allWays.cy.ts:25]: why can't we use data cy atribute here instead of `thead tr th div div:first-child`?

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: mw-webapp/cypress/e2e/allWays.cy.ts (hunk 1 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/allWays.cy.ts]: done

Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/allWays.cy.ts]: @tritonJS826 done

File: mw-webapp/src/logic/waysTable/WaysTable.tsx [hunk 2]
```diff
@@ -40,6 +41,10 @@ export const WaysTable = observer((props: WaysTableProps) => {
   const tableContent = {headerGroup, rowModel};
 
   return (
-    <Table data={tableContent} />
+    <Table
+      data={tableContent}
+      dataCy={allWaysAccessIds.allWaysTable.table}
+      dataCyTr={allWaysAccessIds.allWaysTable.tableTr}
```
Comment (tritonJS826) [inline mw-webapp/src/logic/waysTable/WaysTable.tsx:47]: There should be only one prop for testing purposes (check confirm component)

File: mw-webapp/src/component/table/Table.tsx [hunk 3]
```diff
@@ -50,6 +50,10 @@ interface TableProps<T> {
    */
   dataCy?: string;
 
+  /**
+   * Data attribute for cypress testing
+   */
+  dataCyTr?: string;
```
Comment (tritonJS826) [inline mw-webapp/src/component/table/Table.tsx:56]: let's use dataCy

File: mw-webapp/cypress/e2e/allWays.cy.ts [hunk 4]
```diff
@@ -8,6 +8,24 @@ describe('All Ways scope tests', () => {
...
+    cy.clearAllStorage();
+  });
+
+  it('NoAuth_AllWays_SelectTableView', () => {
+    const tableHeadersEn = Object.values(allWaysPageContent.waysTable.columns).map(column => column.en);
+    tableHeadersEn.push(allWaysPageData.starSimbol);
+
+    allWaysSelectors.filterViewBlock.getTableViewButton().click();
+
+    allWaysSelectors.allWaysTable.getTableTr().find('th').each((el, index) => {
```
Comment (tritonJS826) [inline mw-webapp/cypress/e2e/allWays.cy.ts:25]: I see you added data attribute for tr, lets add to to th too

File: mw-webapp/cypress/e2e/allWays.cy.ts [hunk 5]
```diff
@@ -8,6 +8,24 @@ describe('All Ways scope tests', () => {
...
+    navigationMenuSelectors.menuItemLinks.getAllWaysItemLink().click();
   });
 
+  afterEach(() => {
+    cy.clearAllStorage();
+  });
+
+  it('NoAuth_AllWays_SelectTableView', () => {
+    const tableHeadersEn = Object.values(allWaysPageContent.waysTable.columns).map(column => column.en);
+    tableHeadersEn.push(allWaysPageData.starSimbol);
```
Comment (Ekaterina1994) [inline mw-webapp/cypress/e2e/allWays.cy.ts:21]: => starSymbol

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: mw-webapp/cypress/e2e/allWays.cy.ts (hunk 5 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/allWays.cy.ts]: edit

File: mw-webapp/src/component/table/Table.tsx (hunk 3 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/src/component/table/Table.tsx]: done

Comment (EkaterinaSokolova-V) [reply mw-webapp/src/component/table/Table.tsx]: done

File: mw-webapp/cypress/e2e/allWays.cy.ts (hunk 4 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/allWays.cy.ts]: done

**Review by tritonJS826 — APPROVED** [review]

**Pull request merged**